File: ios/EventEmitterModule.m
//
//  EventEmitterModule.m
//  Lab8App
//
//  Created by Jesus Eduardo Oliva Viscencio on 06/12/24.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(EventEmitterModule, NSObject)

RCT_EXTERN_METHOD(sendEventToJavaScript)

@end

File: ios/EventEmitterModule.swift
//
//  EventEmitterModule.swift
//  Lab8App
//
//  Created by Jesus Eduardo Oliva Viscencio on 06/12/24.
//

import Foundation
import React
 
@objc(EventEmitterModule)
class EventEmitterModule: RCTEventEmitter {
 
    override static func requiresMainQueueSetup() -> Bool {
        return false
    }
 
    override func supportedEvents() -> [String]! {
        return ["onCustomEvent"]
    }
 
    @objc
    func sendEventToJavaScript() {
        sendEvent(withName: "onCustomEvent", body: ["message": "Hello from Swift!"])
    }
}
